api/services/trip/src/actions/PublicStatsAction.ts

```diff
@@ -0,0 +1,26 @@
+// tslint:disable:variable-name
+import { Action } from '@ilos/core';
+import { handler } from '@ilos/common';
+
+import { ParamsInterface, ResultInterface } from '../shared/trip/stats.contract';
+import { TripRepositoryProvider } from '../providers/TripRepositoryProvider';
+import { StatCacheRepositoryProviderInterfaceResolver } from '../interfaces/StatCacheRepositoryProviderInterface';
+import { ApiGraphTimeMode } from '../shared/trip/common/interfaces/ApiGraphTimeMode';
+
+@handler({
+  service: 'trip',
+  method: 'publicStats',
+})
+export class PublicStatsAction extends Action {
+  constructor(private pg: TripRepositoryProvider, private cache: StatCacheRepositoryProviderInterfaceResolver) {
+    super();
+  }
+
+  public async handle(): Promise<ResultInterface> {
+    const params: ParamsInterface = {
+      group_by: ApiGraphTimeMode.All,
+    };
+
+    return (await this.cache.getOrBuild(async () => this.pg.stats(params), params)) || [];
```

Comment:
quelle est la différence avec l'action 'stats' ?

Reply:
Les params sont forcés et y'a pas de check sur les permissions.
1. publicStats => pas authentifié, pas de filtre custom possible (même chose pour tlm)
2. stats (dashboard) => authentifié, filtres custom, besoin de checks sur les permissions un peu plus poussés.

J'ai peur que si on merge les 2, ça rajoute de la complexité aux middlewares de StatsActions ? Et je pense que c'est une bonne pratique de séparer les 2. What do you think ?

Reply:
ok pour moi.